Source/Canary/Mqtt5Canary/Mqtt5Canary.swift

```diff
@@ -269,9 +302,13 @@ class Mqtt5CanaryTestContext: @unchecked Sendable {
     // If there is one client left
     if mqtt5CanaryClients.count > 1 {
       let canaryClient = try getCanaryClient(clientIndex)
-      try canaryClient.client!.stop()
+      if let _client = await canaryClient.client {
+        try _client.stop()
+      } else {
+        throw CanaryTestError.InvalidArgument
+      }
       await statistic.incrementTotalOperation()
-      canaryClient.client = nil
+      await canaryClient.resetClient()
```

Comment:
following this we call `mqtt5CanaryClients.remove(at: mqtt5CanaryClients.index(mqtt5CanaryClients.startIndex, offsetBy: clientIndex))`. Would it be safer to get the `canaryClient.clientId` after we get a client using the clientIndex above and then remove the client using the key instead of an index that could potentially shift around with `mqtt5CanaryClients.removeValue(forKey: clientId)`? In general, since mqtt5CanaryClients is established as `[String: Mqtt5CanaryClient]` I feel a little uneasy about using an index instead of the actual key everywhere. Maybe it's fine and there are no race conditions to be worried about here since these operations SHOULD be happening synchronously but using clientIds instead of index would make sure we're not doing things unintentionally